dataprofiler/profilers/profile_builder.py

```diff
@@ -1927,18 +1954,39 @@ def _update_row_statistics(
                 "Cannot calculate row statistics on data that is" "not a DataFrame"
             )
 
-        self.total_samples += len(data)
-        try:
-            self.hashed_row_dict.update(
-                dict.fromkeys(pd.util.hash_pandas_object(data, index=False), True)
-            )
-        except TypeError:
-            self.hashed_row_dict.update(
-                dict.fromkeys(
-                    pd.util.hash_pandas_object(data.astype(str), index=False), True
-                )
-            )
+        if not self.options.row_statistics.is_enabled:
+            return
 
+        if self.options.row_statistics.unique_count.is_enabled:
+            self.total_samples += len(data)
```

Comment:
` self.total_samples += len(data)` this is actually dangerous to be in here, in fact, dangerous to be in `row_statistics`. Likely should move this outside as `self.total_samples` is critical to keep